raftwal/fsm.go

```diff
@@ -141,6 +146,18 @@ func (fsm *BalloonFSM) Apply(l *raft.Log) interface{} {
 		}
 		return &fsmAddResponse{error: fmt.Errorf("state already applied!: %+v -> %+v", fsm.state, newState)}
 
+	case commands.AddBulkEventCommandType:
+		var cmd commands.AddBulkEventCommand
+		if err := commands.Decode(buf[1:], &cmd); err != nil {
+			return &fsmAddBulkResponse{error: err}
+		}
+		// INFO: after applying a bulk there will be a jump in term version due to balloon version mapping.
+		newState := &fsmState{l.Index, l.Term, fsm.balloon.Version()}
```

Comment:
The new fsmState should include the version that the balloon should have after applying the bulk, so instead `fsm.balloon.Version()` it should be `fsm.balloon.Version() + len(cmd.Events)-1`